### PR TITLE
[SYCL] Add forward declarations for Subgroup Shuffle functions

### PR DESCRIPTION
**Pull request by maksimsab**
The main reason for the change is being able to call Generic overloading of Shuffle from Vector overloading of Shuffle.

### CODE REVIEW

**Comment by maksimsab** [comment]
/verify with https://github.com/intel/llvm-test-suite/pull/1618

**Review by Pennycook — APPROVED** [review]

File: sycl/include/sycl/detail/spirv.hpp [hunk 1]
```diff
@@ -537,6 +537,48 @@ using EnableIfVectorShuffle =
     std::enable_if_t<detail::is_vector_arithmetic<T>::value, T>;
 #endif // ifndef __NVPTX__
 
+// Bitcast shuffles can be implemented using a single SubgroupShuffle
+// intrinsic, but require type-punning via an appropriate integer type
+#ifndef __NVPTX__
+template <typename T>
+using EnableIfBitcastShuffle =
+    detail::enable_if_t<!detail::is_arithmetic<T>::value &&
```
Comment (dm-vodopyanov) [inline sycl/include/sycl/detail/spirv.hpp:545]: nit: `std::enable_if_t` here and below

**Review by dm-vodopyanov — APPROVED** [review]

File: sycl/include/sycl/detail/spirv.hpp (hunk 1 above)
Comment (aelovikov-intel) [reply sycl/include/sycl/detail/spirv.hpp:545]: Same for `_v` version instead of `::value`.

Comment (maksimsab) [reply sycl/include/sycl/detail/spirv.hpp]: done

**Comment by maksimsab** [comment]
Hi @intel/llvm-gatekeepers
It looks like we can merge it.

**Pull request merged**